controller-server/src/main/java/com/yahoo/vespa/hosted/controller/maintenance/HostSwitchUpdater.java

```diff
@@ -31,14 +36,22 @@ protected boolean maintain() {
         Map<String, NodeEntity> nodeEntities = controller().serviceRegistry().entityService().listNodes().stream()
                                                            .collect(Collectors.toMap(NodeEntity::hostname,
                                                                                      Function.identity()));
-        for (var zone : controller().zoneRegistry().zones().controllerUpgraded().all().ids()) {
-            for (var node : nodeRepository.list(zone)) {
-                NodeEntity nodeEntity = nodeEntities.get(node.hostname().value());
-                if (!shouldUpdate(node, nodeEntity)) continue;
-
-                NodeRepositoryNode updatedNode = new NodeRepositoryNode();
-                updatedNode.setSwitchHostname(nodeEntity.switchHostname().get());
-                nodeRepository.patchNode(zone, node.hostname().value(), updatedNode);
+        int nodesUpdated = 0;
+        try {
+            for (var zone : controller().zoneRegistry().zones().controllerUpgraded().all().ids()) {
```

Comment:
Could also limit this to `yahoo` cloud zones...

Reply:
Yes, but that's not available here so would have to use string. 😞